### ISSUE TITLE
Interpretation of Bias value from TreeShap from weighted xgboost model R

### ISSUE THREAD

**Issue by kransom14** [shap/shap#2395]
Hi @slundberg 

I am using weights in an xgboost regression (standard squared loss function). I've noticed that the mean prediction for the training data does not equal the Bias value calculated by TreeShap for the model when the response variable is weighted (but if I do not use weights the values are equal). How to interpret the Bias value for this type of weighted model? Are the shap values still reliable? Below is an example. Note that the bias and mean prediction in the below example are nearly equal, but in my application, the values are highly different. 

```
library(xgboost)

data(iris)

train_label <- iris$Sepal.Length
train_data <- iris[,c(2:4)]
train_m <- as.matrix(train_data)

wts <- c(0.231725156, 0.354273502, 0.486227495, 0.553066575, 0.291290616, 0.084193466, 0.553066575, 0.291290616,
0.683138835, 0.354273502, 0.084193466, 0.419607799, 0.419607799, 0.744239987, 0.000000000, 0.005795098,
0.084193466, 0.231725156, 0.005795098, 0.231725156, 0.084193466, 0.231725156, 0.553066575, 0.231725156,
0.419607799, 0.291290616, 0.291290616, 0.176643136, 0.176643136, 0.486227495, 0.419607799, 0.084193466,
0.176643136, 0.048957845, 0.354273502, 0.291290616, 0.048957845, 0.354273502, 0.683138835, 0.231725156,
0.291290616, 0.619059026, 0.683138835, 0.291290616, 0.231725156, 0.419607799, 0.231725156, 0.553066575,
0.127110568, 0.291290616, 0.366555724, 0.108877671, 0.317665516, 0.048957845, 0.144283968, 0.005795098,
0.077604718, 0.354273502, 0.183344754, 0.176643136, 0.291290616, 0.003423425, 0.013374463, 0.029374260,
0.022469718, 0.225581177, 0.022469718, 0.000000000, 0.050943963, 0.022469718, 0.003423425, 0.029374260,
0.077604718, 0.029374260, 0.108877671, 0.183344754, 0.270514383, 0.225581177, 0.013374463, 0.005795098,
0.048957845, 0.048957845, 0.000000000, 0.013374463, 0.084193466, 0.013374463, 0.225581177, 0.077604718,
0.022469718, 0.048957845, 0.048957845, 0.029374260, 0.000000000, 0.291290616, 0.022469718, 0.005795098,
0.005795098, 0.050943963, 0.231725156, 0.005795098, 0.077604718, 0.000000000, 0.416706153, 0.077604718,
0.144283968, 0.669601720, 0.354273502, 0.518872257, 0.225581177, 0.467637949, 0.144283968, 0.108877671,
0.270514383, 0.005795098, 0.000000000, 0.108877671, 0.144283968, 0.717257541, 0.717257541, 0.013374463,
0.317665516, 0.022469718, 0.717257541, 0.077604718, 0.225581177, 0.467637949, 0.050943963, 0.029374260,
0.108877671, 0.467637949, 0.569930224, 0.805815059, 0.108877671, 0.077604718, 0.029374260, 0.717257541,
0.077604718, 0.108877671, 0.013374463, 0.317665516, 0.225581177, 0.317665516, 0.000000000, 0.270514383,
0.225581177, 0.225581177, 0.077604718, 0.144283968, 0.050943963, 0.003423425)

train.d <- xgb.DMatrix(train_m, label = train_label, weight = wts)
train.d2 <- xgb.DMatrix(train_m, label = train_label)

xgb_mod <- xgb.train(data = train.d, nrounds = 50)
xgb_mod2 <- xgb.train(data = train.d2, nrounds = 50)

# get the shap for training wells for weighted model
s <- predict(xgb_mod, train_m, predcontrib = TRUE, approxcontrib = FALSE)
tot_shap <- rowSums(s) # includes bias column

s2 <- predict(xgb_mod2, train_m, predcontrib = TRUE, approxcontrib = FALSE)
tot_shap2 <- rowSums(s2) # includes bias column

# get the training data predictions
p <- predict(xgb_mod, train_m)
p2 <- predict(xgb_mod2, train_m)

# check that total shap equals the prediction
plot(p, tot_shap) # prediction is equal to total shap
abline(0,1)

mean(p) # mean training data prediction is lower than bias for shap from weighted model
s[1,4] #shap bias

# check that total shap equals the prediction
plot(p2, tot_shap2) # prediction is equal to total shap
abline(0,1)

mean(p2) # mean training data prediction is equal to shap from non-weighted model
s2[1,4] #shap bias
```

**Comment by kransom14** [comment]
Just curious if anyone has had a chance to look into this or if someone may be able to provide some insight or confirmation that the shap values are still reliable for a weighted xgboost model? Weights really improve my model performance and I'd like to be able to use shap to explain the predictions. Thank you in advance.
